cql3/statements/alter_table_statement.cc

```diff
@@ -438,15 +452,26 @@ alter_table_statement::raw_statement::prepare(data_dictionary::database db, cql_
     }
 
     auto ctx = get_prepare_context();
+    auto prepared_attrs = _attrs->prepare(db, keyspace(), column_family());
+    prepared_attrs->fill_prepare_context(ctx);
+
+    auto bound_terms = ctx.bound_variables_size();
+
+    if (bound_terms > 0 && !s) {
+        throw exceptions::invalid_request_exception("Cannot prepare a statement with parameter markers with non-existing schema");
+    }
 
     return std::make_unique<prepared_statement>(::make_shared<alter_table_statement>(
-        ctx.bound_variables_size(),
-        std::move(*_cf_name),
-        _type,
-        std::move(_column_changes),
-        std::move(_properties),
-        std::move(_renames)
-    ));
+                std::move(bound_terms),
+                std::move(*_cf_name),
+                _type,
+                std::move(_column_changes),
+                std::move(_properties),
+                std::move(_renames),
+                std::move(prepared_attrs)
+            ),
+            ctx,
+            std::vector<uint16_t>()); // since alter table is `cql_statement_no_metadata` we can pass empty vector as partition_key_bind_indices
```

Comment:
I didn't understand what this change means. Thanks for writing a comment, but I don't understand what it is saying.

Reply:
`partition_key_bind_indices` is by driver to calculate partition key if bind markers are part of it. That's why it can be simply empty vector here.
Extended the comment